src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,443 @@
+{
+  /* length of the command */
+  unsigned int command_length = 0;
+
+  /* convert the active-frequency to an AR62xx-ID */
+  uint16_t active_frequency_idx = FrequencyToAR62xxId(active_frequency);
+
+  /* convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /* command-start */
+  command[command_length++] = 0xA5;
+
+  /* add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /* set active station */
+  if (active_passive == ACTIVE_STATION)
+  {
+    active_frequency = frequency;
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+  }
+
+  /* set passive station */
+  else
+  {
+    passive_frequency = frequency;
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+  }
+
+  /* setting frequencies */
+  command[command_length++] = 22;
+
+  //add active station
+  uint8_t tmp[2] = {};
+  WriteUnalignedLE16((uint16_t *)(void *)tmp, active_frequency_idx);
+  command[command_length++] = tmp[1];
+  command[command_length++] = tmp[0];
+
+  //add passive station
+  WriteUnalignedLE16((uint16_t *)(void *)tmp, passive_frequency_idx);
+  command[command_length++] = tmp[1];
+  command[command_length++] = tmp[0];
+
+  //add crc
+  WriteUnalignedLE16((uint16_t *)(void *)tmp, CRCBitwise(command, command_length));
+  command[command_length++] = tmp[1];
+  command[command_length++] = tmp[0];
+
+  //return full command
+  return command_length;
+}
+
+bool AR62xxDevice::ParseReceivedData(const uint8_t *data, size_t len)
+{
+  size_t idx = 0;
+  size_t rec_buf_len = 0;
+
+  /* we only need to read the message until record-size = 10 because then it already contains
+   * all data we want to read (active and passive frequency) */
+  size_t rec_bufsize = 10;
+  uint8_t command[rec_bufsize];
+
+  //flag if frequency has been saved
+  bool frequencySaved = false;
+
+  /* loop as long we have data and stop if frequency has been set */
+  while (!frequencySaved && rec_buf_len < rec_bufsize && idx < len)
+  {
+    /* check if we received a header which tells us now data starts */
+    if (data[idx] == 0xA5)
+    {
+      /* reset buffer length */
+      rec_buf_len = 0;
+    }
+
+    /* write command in buffer */
+    command[rec_buf_len++] = (char)data[idx++];
+
+    /* check if record-length == 10 then we can read the frequencies (active and passive frequency of ar62xx)
+     */
+    if (rec_buf_len == rec_bufsize)
+    {
+      size_t command_length = command[2];
+
+      /* generate uint16_t out of commands */
+      u_int8_t tmp[] = {command[command_length + 4], command[command_length + 3]};
+      uint16_t crc_value = ReadUnalignedLE16((const uint16_t *)(const void *)tmp);
+
+      /* generate crc */
+      uint16_t crc = CRCBitwise(command, command_length + 3);
+
+      /* check crc */
+      if (crc == crc_value || crc_value == 0)
```

Comment:
Why disable CRC checks if the declared CRC happens to be zero?

Reply:
solved... please have a look at the new implementation which is completely different from this and fully documented :-)